// baseui/table-grid/styled-components.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledBodyCell = exports.StyledHeadCell = exports.StyledTable = void 0;

var _index = require("../styles/index.js");

var _index2 = require("../table/index.js");

/*
Copyright (c) 2018-2019 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
var StyledTable = (0, _index.withStyle)(_index2.StyledTable, function (props) {
  return {
    display: 'grid',
    gridTemplateColumns: props.$gridTemplateColumns,
    flexDirection: 'unset'
  };
});
exports.StyledTable = StyledTable;
StyledTable.displayName = "StyledTable";
var StyledHeadCell = (0, _index.withStyle)(_index2.StyledHeadCell, function (_ref) {
  var _ref$$sticky = _ref.$sticky,
      $sticky = _ref$$sticky === void 0 ? true : _ref$$sticky,
      $theme = _ref.$theme;
  return {
    backgroundColor: $theme.colors.tableHeadBackgroundColor,
    boxShadow: $theme.lighting.shadow400,
    position: $sticky ? 'sticky' : null,
    top: $sticky ? 0 : null,
    width: 'unset'
  };
});
exports.StyledHeadCell = StyledHeadCell;
StyledHeadCell.displayName = "StyledHeadCell";
var StyledBodyCell = (0, _index.withStyle)(_index2.StyledCell, function (props) {
  return {
    display: 'block',
    flex: 'unset',
    gridColumn: props.$gridColumn || null,
    gridRow: props.$gridRow || null
  };
});
exports.StyledBodyCell = StyledBodyCell;
StyledBodyCell.displayName = "StyledBodyCell";
