// baseui/themes/dark-theme-colors.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _darkThemePrimitives = require("./dark-theme-primitives.js");

/*
Copyright (c) 2018-2019 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
var WHITE = '#FFFFFF';
var _default = {
  colors: {
    // Semantic Colors
    white: WHITE,
    // Font Color
    colorPrimary: _darkThemePrimitives.primitives.mono100,
    colorSecondary: _darkThemePrimitives.primitives.mono200,
    // Background
    background: _darkThemePrimitives.primitives.mono1000,
    backgroundAlt: _darkThemePrimitives.primitives.mono700,
    backgroundInv: _darkThemePrimitives.primitives.mono100,
    // Foreground
    foreground: _darkThemePrimitives.primitives.mono100,
    foregroundAlt: _darkThemePrimitives.primitives.mono300,
    foregroundInv: _darkThemePrimitives.primitives.mono1000,
    // Borders
    border: _darkThemePrimitives.primitives.mono600,
    borderAlt: _darkThemePrimitives.primitives.mono700,
    borderFocus: _darkThemePrimitives.primitives.primary400,
    borderError: _darkThemePrimitives.primitives.negative400,
    // Buttons
    buttonPrimaryFill: _darkThemePrimitives.primitives.primary500,
    buttonPrimaryText: _darkThemePrimitives.primitives.mono100,
    buttonPrimaryHover: _darkThemePrimitives.primitives.primary600,
    buttonPrimaryActive: _darkThemePrimitives.primitives.primary700,
    buttonSecondaryFill: _darkThemePrimitives.primitives.mono600,
    buttonSecondaryText: _darkThemePrimitives.primitives.mono100,
    buttonSecondaryHover: _darkThemePrimitives.primitives.mono500,
    buttonSecondaryActive: _darkThemePrimitives.primitives.mono400,
    buttonTertiaryFill: _darkThemePrimitives.primitives.mono800,
    buttonTertiaryText: _darkThemePrimitives.primitives.mono100,
    buttonTertiaryHover: _darkThemePrimitives.primitives.mono700,
    buttonTertiaryActive: _darkThemePrimitives.primitives.mono600,
    buttonTertiarySelectedText: _darkThemePrimitives.primitives.mono100,
    buttonTertiarySelectedFill: _darkThemePrimitives.primitives.primary500,
    buttonMinimalFill: 'transparent',
    buttonMinimalText: _darkThemePrimitives.primitives.primary400,
    buttonMinimalHover: _darkThemePrimitives.primitives.mono800,
    buttonMinimalActive: _darkThemePrimitives.primitives.mono700,
    buttonDisabledFill: _darkThemePrimitives.primitives.mono700,
    buttonDisabledText: _darkThemePrimitives.primitives.mono500,
    // Breadcrumbs
    breadcrumbsText: _darkThemePrimitives.primitives.mono100,
    breadcrumbsSeparatorFill: _darkThemePrimitives.primitives.mono200,
    // Datepicker
    datepickerBackground: _darkThemePrimitives.primitives.mono600,
    datepickerDayFont: WHITE,
    datepickerDayFontDisabled: _darkThemePrimitives.primitives.mono100,
    datepickerDayPseudoSelected: _darkThemePrimitives.primitives.mono500,
    datepickerDayPseudoHighlighted: _darkThemePrimitives.primitives.mono600,
    // FileUploader
    fileUploaderBackgroundColor: _darkThemePrimitives.primitives.mono700,
    fileUploaderBackgroundColorActive: _darkThemePrimitives.primitives.mono600,
    fileUploaderBorderColorActive: _darkThemePrimitives.primitives.primary400,
    fileUploaderBorderColorDefault: _darkThemePrimitives.primitives.mono500,
    fileUploaderMessageColor: _darkThemePrimitives.primitives.mono100,
    // Links
    linkText: _darkThemePrimitives.primitives.primary300,
    linkVisited: _darkThemePrimitives.primitives.primary300,
    linkHover: _darkThemePrimitives.primitives.primary400,
    linkActive: _darkThemePrimitives.primitives.primary400,
    // List
    listHeaderFill: _darkThemePrimitives.primitives.mono600,
    listBodyFill: _darkThemePrimitives.primitives.mono700,
    listIconFill: _darkThemePrimitives.primitives.mono100,
    listBorder: _darkThemePrimitives.primitives.mono500,
    // ProgressSteps
    progressStepsIconActiveFill: _darkThemePrimitives.primitives.mono100,
    // Notification
    notificationPrimaryBackground: _darkThemePrimitives.primitives.primary700,
    notificationPrimaryText: _darkThemePrimitives.primitives.primary200,
    notificationPositiveBackground: _darkThemePrimitives.primitives.positive700,
    notificationPositiveText: _darkThemePrimitives.primitives.positive200,
    notificationWarningBackground: _darkThemePrimitives.primitives.warning700,
    notificationWarningText: _darkThemePrimitives.primitives.warning200,
    notificationNegativeBackground: _darkThemePrimitives.primitives.negative700,
    notificationNegativeText: _darkThemePrimitives.primitives.negative200,
    // Tag
    tagSolidRampUnit: '500',
    tagSolidHoverRampUnit: '500',
    tagSolidActiveRampUnit: '400',
    tagSolidDisabledRampUnit: '700',
    tagSolidFontRampUnit: '100',
    tagSolidFontHoverRampUnit: '100',
    tagLightRampUnit: '700',
    tagLightHoverRampUnit: '700',
    tagLightActiveRampUnit: '600',
    tagLightDisabledRampUnit: '700',
    tagLightFontRampUnit: '100',
    tagLightFontHoverRampUnit: '100',
    tagOutlinedRampUnit: '500',
    tagOutlinedHoverRampUnit: '400',
    tagOutlinedActiveRampUnit: '300',
    tagOutlinedDisabledRampUnit: '700',
    tagOutlinedFontRampUnit: '200',
    tagOutlinedFontHoverRampUnit: '700',
    tagFontDisabledRampUnit: '500',
    tagNeutralSolidBackground: _darkThemePrimitives.primitives.mono600,
    tagNeutralSolidHover: _darkThemePrimitives.primitives.mono600,
    tagNeutralSolidActive: _darkThemePrimitives.primitives.mono500,
    tagNeutralSolidDisabled: _darkThemePrimitives.primitives.mono700,
    tagNeutralSolidFont: _darkThemePrimitives.primitives.mono200,
    tagNeutralSolidFontHover: _darkThemePrimitives.primitives.mono200,
    tagNeutralLightBackground: _darkThemePrimitives.primitives.mono800,
    tagNeutralLightHover: _darkThemePrimitives.primitives.mono800,
    tagNeutralLightActive: _darkThemePrimitives.primitives.mono700,
    tagNeutralLightDisabled: _darkThemePrimitives.primitives.mono700,
    tagNeutralLightFont: _darkThemePrimitives.primitives.mono200,
    tagNeutralLightFontHover: _darkThemePrimitives.primitives.mono200,
    tagNeutralOutlinedBackground: _darkThemePrimitives.primitives.mono600,
    tagNeutralOutlinedHover: _darkThemePrimitives.primitives.mono500,
    tagNeutralOutlinedActive: _darkThemePrimitives.primitives.mono400,
    tagNeutralOutlinedDisabled: _darkThemePrimitives.primitives.mono700,
    tagNeutralOutlinedFont: _darkThemePrimitives.primitives.mono200,
    tagNeutralOutlinedFontHover: _darkThemePrimitives.primitives.mono900,
    tagNeutralFontDisabled: _darkThemePrimitives.primitives.mono500,
    tagPrimarySolidBackground: _darkThemePrimitives.primitives.primary500,
    tagPrimarySolidHover: _darkThemePrimitives.primitives.primary500,
    tagPrimarySolidActive: _darkThemePrimitives.primitives.primary400,
    tagPrimarySolidDisabled: _darkThemePrimitives.primitives.primary700,
    tagPrimarySolidFont: _darkThemePrimitives.primitives.primary100,
    tagPrimarySolidFontHover: _darkThemePrimitives.primitives.primary100,
    tagPrimaryLightBackground: _darkThemePrimitives.primitives.primary700,
    tagPrimaryLightHover: _darkThemePrimitives.primitives.primary700,
    tagPrimaryLightActive: _darkThemePrimitives.primitives.primary600,
    tagPrimaryLightDisabled: _darkThemePrimitives.primitives.primary700,
    tagPrimaryLightFont: _darkThemePrimitives.primitives.primary100,
    tagPrimaryLightFontHover: _darkThemePrimitives.primitives.primary100,
    tagPrimaryOutlinedBackground: _darkThemePrimitives.primitives.primary500,
    tagPrimaryOutlinedHover: _darkThemePrimitives.primitives.primary400,
    tagPrimaryOutlinedActive: _darkThemePrimitives.primitives.primary300,
    tagPrimaryOutlinedDisabled: _darkThemePrimitives.primitives.primary700,
    tagPrimaryOutlinedFont: _darkThemePrimitives.primitives.primary200,
    tagPrimaryOutlinedFontHover: _darkThemePrimitives.primitives.primary700,
    tagPrimaryFontDisabled: _darkThemePrimitives.primitives.primary500,
    tagPositiveSolidBackground: _darkThemePrimitives.primitives.positive500,
    tagPositiveSolidHover: _darkThemePrimitives.primitives.positive500,
    tagPositiveSolidActive: _darkThemePrimitives.primitives.positive400,
    tagPositiveSolidDisabled: _darkThemePrimitives.primitives.positive700,
    tagPositiveSolidFont: _darkThemePrimitives.primitives.positive100,
    tagPositiveSolidFontHover: _darkThemePrimitives.primitives.positive100,
    tagPositiveLightBackground: _darkThemePrimitives.primitives.positive700,
    tagPositiveLightHover: _darkThemePrimitives.primitives.positive700,
    tagPositiveLightActive: _darkThemePrimitives.primitives.positive600,
    tagPositiveLightDisabled: _darkThemePrimitives.primitives.positive700,
    tagPositiveLightFont: _darkThemePrimitives.primitives.positive100,
    tagPositiveLightFontHover: _darkThemePrimitives.primitives.positive100,
    tagPositiveOutlinedBackground: _darkThemePrimitives.primitives.positive500,
    tagPositiveOutlinedHover: _darkThemePrimitives.primitives.positive400,
    tagPositiveOutlinedActive: _darkThemePrimitives.primitives.positive300,
    tagPositiveOutlinedDisabled: _darkThemePrimitives.primitives.positive700,
    tagPositiveOutlinedFont: _darkThemePrimitives.primitives.positive200,
    tagPositiveOutlinedFontHover: _darkThemePrimitives.primitives.positive700,
    tagPositiveFontDisabled: _darkThemePrimitives.primitives.positive500,
    tagWarningSolidBackground: _darkThemePrimitives.primitives.warning500,
    tagWarningSolidHover: _darkThemePrimitives.primitives.warning500,
    tagWarningSolidActive: _darkThemePrimitives.primitives.warning400,
    tagWarningSolidDisabled: _darkThemePrimitives.primitives.warning700,
    tagWarningSolidFont: _darkThemePrimitives.primitives.warning100,
    tagWarningSolidFontHover: _darkThemePrimitives.primitives.warning100,
    tagWarningLightBackground: _darkThemePrimitives.primitives.warning700,
    tagWarningLightHover: _darkThemePrimitives.primitives.warning700,
    tagWarningLightActive: _darkThemePrimitives.primitives.warning600,
    tagWarningLightDisabled: _darkThemePrimitives.primitives.warning700,
    tagWarningLightFont: _darkThemePrimitives.primitives.warning100,
    tagWarningLightFontHover: _darkThemePrimitives.primitives.warning100,
    tagWarningOutlinedBackground: _darkThemePrimitives.primitives.warning500,
    tagWarningOutlinedHover: _darkThemePrimitives.primitives.warning400,
    tagWarningOutlinedActive: _darkThemePrimitives.primitives.warning300,
    tagWarningOutlinedDisabled: _darkThemePrimitives.primitives.warning700,
    tagWarningOutlinedFont: _darkThemePrimitives.primitives.warning200,
    tagWarningOutlinedFontHover: _darkThemePrimitives.primitives.warning700,
    tagWarningFontDisabled: _darkThemePrimitives.primitives.warning500,
    tagNegativeSolidBackground: _darkThemePrimitives.primitives.negative500,
    tagNegativeSolidHover: _darkThemePrimitives.primitives.negative500,
    tagNegativeSolidActive: _darkThemePrimitives.primitives.negative400,
    tagNegativeSolidDisabled: _darkThemePrimitives.primitives.negative700,
    tagNegativeSolidFont: _darkThemePrimitives.primitives.negative100,
    tagNegativeSolidFontHover: _darkThemePrimitives.primitives.negative100,
    tagNegativeLightBackground: _darkThemePrimitives.primitives.negative700,
    tagNegativeLightHover: _darkThemePrimitives.primitives.negative700,
    tagNegativeLightActive: _darkThemePrimitives.primitives.negative600,
    tagNegativeLightDisabled: _darkThemePrimitives.primitives.negative700,
    tagNegativeLightFont: _darkThemePrimitives.primitives.negative100,
    tagNegativeLightFontHover: _darkThemePrimitives.primitives.negative100,
    tagNegativeOutlinedBackground: _darkThemePrimitives.primitives.negative500,
    tagNegativeOutlinedHover: _darkThemePrimitives.primitives.negative400,
    tagNegativeOutlinedActive: _darkThemePrimitives.primitives.negative300,
    tagNegativeOutlinedDisabled: _darkThemePrimitives.primitives.negative700,
    tagNegativeOutlinedFont: _darkThemePrimitives.primitives.negative200,
    tagNegativeOutlinedFontHover: _darkThemePrimitives.primitives.negative700,
    tagNegativeFontDisabled: _darkThemePrimitives.primitives.negative500,
    // Table
    tableHeadBackgroundColor: _darkThemePrimitives.primitives.mono700,
    tableBackground: _darkThemePrimitives.primitives.mono800,
    tableStripedBackground: _darkThemePrimitives.primitives.mono700,
    tableFilter: _darkThemePrimitives.primitives.mono400,
    tableFilterHeading: _darkThemePrimitives.primitives.mono300,
    tableFilterBackground: _darkThemePrimitives.primitives.mono700,
    tableFilterFooterBackground: _darkThemePrimitives.primitives.mono800,
    // Tick
    tickFill: _darkThemePrimitives.primitives.mono1000,
    tickFillHover: _darkThemePrimitives.primitives.mono800,
    tickFillActive: _darkThemePrimitives.primitives.mono600,
    tickFillSelected: _darkThemePrimitives.primitives.primary500,
    tickFillSelectedHover: _darkThemePrimitives.primitives.primary600,
    tickFillSelectedHoverActive: _darkThemePrimitives.primitives.primary700,
    tickFillError: _darkThemePrimitives.primitives.negative700,
    tickFillErrorHover: _darkThemePrimitives.primitives.negative600,
    tickFillErrorHoverActive: _darkThemePrimitives.primitives.negative500,
    tickFillErrorSelected: _darkThemePrimitives.primitives.negative500,
    tickFillErrorSelectedHover: _darkThemePrimitives.primitives.negative600,
    tickFillErrorSelectedHoverActive: _darkThemePrimitives.primitives.negative700,
    tickFillDisabled: _darkThemePrimitives.primitives.mono700,
    tickBorder: _darkThemePrimitives.primitives.mono300,
    tickBorderError: _darkThemePrimitives.primitives.negative400,
    tickMarkFill: _darkThemePrimitives.primitives.mono100,
    tickMarkFillDisabled: _darkThemePrimitives.primitives.mono400,
    // Slider/Toggle
    sliderTrackFill: _darkThemePrimitives.primitives.mono600,
    sliderTrackFillHover: _darkThemePrimitives.primitives.mono500,
    sliderTrackFillActive: _darkThemePrimitives.primitives.mono400,
    sliderTrackFillSelected: _darkThemePrimitives.primitives.primary500,
    sliderTrackFillSelectedActive: _darkThemePrimitives.primitives.primary600,
    sliderTrackFillSelectedHover: _darkThemePrimitives.primitives.primary700,
    sliderTrackFillDisabled: _darkThemePrimitives.primitives.mono700,
    sliderHandleFill: _darkThemePrimitives.primitives.mono300,
    sliderHandleFillHover: _darkThemePrimitives.primitives.mono300,
    sliderHandleFillActive: _darkThemePrimitives.primitives.mono300,
    sliderHandleFillSelected: _darkThemePrimitives.primitives.primary500,
    sliderHandleFillSelectedHover: _darkThemePrimitives.primitives.primary600,
    sliderHandleFillSelectedActive: _darkThemePrimitives.primitives.primary700,
    sliderHandleFillDisabled: _darkThemePrimitives.primitives.mono600,
    sliderHandleInnerFill: _darkThemePrimitives.primitives.mono300,
    sliderHandleInnerFillDisabled: _darkThemePrimitives.primitives.mono500,
    sliderHandleInnerFillSelectedHover: _darkThemePrimitives.primitives.primary600,
    sliderHandleInnerFillSelectedActive: _darkThemePrimitives.primitives.primary700,
    sliderBorder: WHITE,
    sliderBorderHover: WHITE,
    sliderBorderDisabled: _darkThemePrimitives.primitives.mono400,
    // Input
    inputFill: _darkThemePrimitives.primitives.mono600,
    inputFillActive: _darkThemePrimitives.primitives.mono500,
    inputFillError: _darkThemePrimitives.primitives.negative700,
    inputFillDisabled: _darkThemePrimitives.primitives.mono800,
    inputFillPositive: _darkThemePrimitives.primitives.positive700,
    inputTextDisabled: _darkThemePrimitives.primitives.mono500,
    inputBorderError: _darkThemePrimitives.primitives.negative400,
    inputBorderPositive: _darkThemePrimitives.primitives.positive400,
    inputEnhancerFill: _darkThemePrimitives.primitives.mono500,
    inputEnhancerFillDisabled: _darkThemePrimitives.primitives.mono700,
    inputEnhancerTextDisabled: _darkThemePrimitives.primitives.mono500,
    // Menu
    menuFill: _darkThemePrimitives.primitives.mono600,
    menuFillHover: _darkThemePrimitives.primitives.mono700,
    menuFontDefault: _darkThemePrimitives.primitives.mono300,
    menuFontDisabled: _darkThemePrimitives.primitives.mono400,
    menuFontHighlighted: _darkThemePrimitives.primitives.mono200,
    menuFontSelected: WHITE,
    // Pagination
    paginationTriangleDown: _darkThemePrimitives.primitives.mono100,
    // Header navigation
    headerNavigationFill: _darkThemePrimitives.primitives.mono700,
    // Tab
    tabBarFill: _darkThemePrimitives.primitives.mono1000,
    tabColor: _darkThemePrimitives.primitives.mono300,
    // Spinner
    spinnerTrackFill: _darkThemePrimitives.primitives.mono100,
    // Progress bar
    progressbarTrackFill: _darkThemePrimitives.primitives.mono100,
    // Tooltip
    tooltipBackground: _darkThemePrimitives.primitives.mono200,
    tooltipText: _darkThemePrimitives.primitives.mono1000
  }
};
exports.default = _default;
