// baseui/card/styled-components.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Title = exports.Thumbnail = exports.Root = exports.HeaderImage = exports.Contents = exports.Body = exports.Action = void 0;

var _index = require("../styles/index.js");

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var Action = (0, _index.styled)('div', function (_ref) {
  var typography = _ref.$theme.typography;
  return _objectSpread({}, typography.font450);
});
exports.Action = Action;
Action.displayName = "Action";
var Body = (0, _index.styled)('div', function (_ref2) {
  var $theme = _ref2.$theme;
  var colors = $theme.colors,
      sizing = $theme.sizing,
      typography = $theme.typography;
  return _objectSpread({
    marginBottom: sizing.scale600,
    color: colors.foregroundAlt
  }, typography.font300);
});
exports.Body = Body;
Body.displayName = "Body";
var Contents = (0, _index.styled)('div', function (_ref3) {
  var $theme = _ref3.$theme;
  var sizing = $theme.sizing;
  return {
    marginLeft: sizing.scale800,
    marginTop: sizing.scale800,
    marginRight: sizing.scale800,
    marginBottom: sizing.scale800
  };
});
exports.Contents = Contents;
Contents.displayName = "Contents";
var HeaderImage = (0, _index.styled)('img', function (_ref4) {
  var borders = _ref4.$theme.borders;
  return {
    borderTopLeftRadius: borders.surfaceBorderRadius,
    borderTopRightRadius: borders.surfaceBorderRadius,
    objectFit: 'contain',
    maxWidth: '100%'
  };
});
exports.HeaderImage = HeaderImage;
HeaderImage.displayName = "HeaderImage";
var Root = (0, _index.styled)('div', function (_ref5) {
  var $theme = _ref5.$theme;
  var borders = $theme.borders,
      lighting = $theme.lighting,
      colors = $theme.colors;
  return _objectSpread({}, borders.border300, {
    boxShadow: lighting.shadow400,
    borderTopLeftRadius: borders.surfaceBorderRadius,
    borderTopRightRadius: borders.surfaceBorderRadius,
    borderBottomLeftRadius: borders.surfaceBorderRadius,
    borderBottomRightRadius: borders.surfaceBorderRadius,
    backgroundColor: colors.backgroundAlt
  });
});
exports.Root = Root;
Root.displayName = "Root";
var Thumbnail = (0, _index.styled)('img', function (props) {
  var _props$$theme = props.$theme,
      borders = _props$$theme.borders,
      sizing = _props$$theme.sizing;
  return _objectSpread({
    float: 'right',
    height: sizing.scale2400,
    width: sizing.scale2400,
    objectFit: 'cover',
    borderTopLeftRadius: borders.surfaceBorderRadius,
    borderTopRightRadius: borders.surfaceBorderRadius,
    borderBottomLeftRadius: borders.surfaceBorderRadius,
    borderBottomRightRadius: borders.surfaceBorderRadius
  }, borders.border200, {
    margin: "0 0 ".concat(sizing.scale500, " ").concat(sizing.scale500)
  });
});
exports.Thumbnail = Thumbnail;
Thumbnail.displayName = "Thumbnail";
var Title = (0, _index.styled)('h1', function (_ref6) {
  var $theme = _ref6.$theme;
  var colors = $theme.colors,
      sizing = $theme.sizing,
      typography = $theme.typography;
  return _objectSpread({}, typography.font500, {
    color: colors.foreground,
    fontWeight: 500,
    marginLeft: 0,
    marginTop: 0,
    marginRight: 0,
    marginBottom: sizing.scale300,
    paddingLeft: 0,
    paddingTop: 0,
    paddingRight: 0,
    paddingBottom: 0
  });
});
exports.Title = Title;
Title.displayName = "Title";
