// baseui/es/header-navigation/styled-components.js
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

/*
Copyright (c) 2018-2019 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
import { styled } from '../styles/index.js';
import { ALIGN } from './constants.js';
export var Root = styled('nav', function (props) {
  var $theme = props.$theme;
  var scale500 = $theme.sizing.scale500,
      font400 = $theme.typography.font400,
      border = $theme.colors.border;
  return _objectSpread({}, font400, {
    display: 'flex',
    paddingBottom: scale500,
    paddingTop: scale500,
    borderBottom: "1px solid ".concat(border),
    backgroundColor: $theme.colors.headerNavigationFill
  });
});
Root.displayName = "Root";
export var NavigationItem = styled('div', function (props) {
  var $theme = props.$theme;
  var scale800 = $theme.sizing.scale800;
  return {
    alignSelf: 'center',
    paddingLeft: scale800
  };
});
NavigationItem.displayName = "NavigationItem";
export var NavigationList = styled('div', function (props) {
  var $align = props.$align,
      $theme = props.$theme;
  var scale800 = $theme.sizing.scale800;
  return {
    display: 'flex',
    ':first-child': {
      padding: 0
    },
    ':last-child': {
      padding: 0
    },
    flex: $align === ALIGN.right || $align === ALIGN.left ? 'none' : 1,
    paddingLeft: scale800,
    paddingRight: scale800,
    justifySelf: $align,
    justifyContent: $align
  };
});
NavigationList.displayName = "NavigationList";
