// baseui/header-navigation/styled-components.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.NavigationList = exports.NavigationItem = exports.Root = void 0;

var _index = require("../styles/index.js");

var _constants = require("./constants.js");

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var Root = (0, _index.styled)('nav', function (props) {
  var $theme = props.$theme;
  var scale500 = $theme.sizing.scale500,
      font400 = $theme.typography.font400,
      border = $theme.colors.border;
  return _objectSpread({}, font400, {
    display: 'flex',
    paddingBottom: scale500,
    paddingTop: scale500,
    borderBottom: "1px solid ".concat(border),
    backgroundColor: $theme.colors.headerNavigationFill
  });
});
exports.Root = Root;
Root.displayName = "Root";
var NavigationItem = (0, _index.styled)('div', function (props) {
  var $theme = props.$theme;
  var scale800 = $theme.sizing.scale800;
  return {
    alignSelf: 'center',
    paddingLeft: scale800
  };
});
exports.NavigationItem = NavigationItem;
NavigationItem.displayName = "NavigationItem";
var NavigationList = (0, _index.styled)('div', function (props) {
  var $align = props.$align,
      $theme = props.$theme;
  var scale800 = $theme.sizing.scale800;
  return {
    display: 'flex',
    ':first-child': {
      padding: 0
    },
    ':last-child': {
      padding: 0
    },
    flex: $align === _constants.ALIGN.right || $align === _constants.ALIGN.left ? 'none' : 1,
    paddingLeft: scale800,
    paddingRight: scale800,
    justifySelf: $align,
    justifyContent: $align
  };
});
exports.NavigationList = NavigationList;
NavigationList.displayName = "NavigationList";
