// baseui/select/styled-components.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledSearchIcon = exports.getLoadingIconStyles = exports.StyledClearIcon = exports.StyledSelectArrow = exports.StyledIconsContainer = exports.StyledInputSizer = exports.StyledInput = exports.StyledInputContainer = exports.StyledSingleValue = exports.StyledPlaceholder = exports.StyledValueContainer = exports.StyledControlContainer = exports.StyledRoot = exports.StyledOptionContent = exports.StyledDropdownListItem = exports.StyledDropdown = exports.StyledDropdownContainer = void 0;

var _index = require("../styles/index.js");

var _constants = require("./constants.js");

var _index2 = require("../menu/index.js");

var _util = require("../styles/util.js");

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function getFont() {
  var _SIZE$compact$SIZE$de;

  var size = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : _constants.SIZE.default;
  var typography = arguments.length > 1 ? arguments[1] : undefined;
  return (_SIZE$compact$SIZE$de = {}, _defineProperty(_SIZE$compact$SIZE$de, _constants.SIZE.compact, typography.font300), _defineProperty(_SIZE$compact$SIZE$de, _constants.SIZE.default, typography.font400), _defineProperty(_SIZE$compact$SIZE$de, _constants.SIZE.large, typography.font460), _SIZE$compact$SIZE$de)[size];
}

function getControlPadding(props, emptyValue) {
  var _SIZE$compact, _SIZE$default, _SIZE$large, _SIZE$compact$SIZE$de2;

  var $theme = props.$theme,
      sizing = props.$theme.sizing,
      _props$$size = props.$size,
      $size = _props$$size === void 0 ? _constants.SIZE.default : _props$$size,
      $type = props.$type,
      $multi = props.$multi;
  var isSearch = $type === _constants.TYPE.search;
  var paddingLeft = isSearch ? sizing.scale1000 : sizing.scale500;
  return (_SIZE$compact$SIZE$de2 = {}, _defineProperty(_SIZE$compact$SIZE$de2, _constants.SIZE.compact, (_SIZE$compact = {
    // `sizing.scale0` based on the multi value component (Tag) top and bottom margin
    paddingTop: $multi && !emptyValue ? "calc(".concat(sizing.scale200, " - ").concat(sizing.scale0, ")") : sizing.scale200,
    paddingBottom: $multi && !emptyValue ? "calc(".concat(sizing.scale200, " - ").concat(sizing.scale0, ")") : sizing.scale200
  }, _defineProperty(_SIZE$compact, $theme.direction === 'rtl' ? 'paddingRight' : 'paddingLeft', $multi && !emptyValue ? "calc(".concat(paddingLeft, " - ").concat(sizing.scale0, ")") : paddingLeft), _defineProperty(_SIZE$compact, $theme.direction === 'rtl' ? 'paddingLeft' : 'paddingRight', '0'), _SIZE$compact)), _defineProperty(_SIZE$compact$SIZE$de2, _constants.SIZE.default, (_SIZE$default = {
    // `sizing.scale0` based on the multi value component (Tag) top and bottom margin
    paddingTop: $multi && !emptyValue ? "calc(".concat(sizing.scale400, " - ").concat(sizing.scale0, ")") : sizing.scale400,
    paddingBottom: $multi && !emptyValue ? "calc(".concat(sizing.scale400, " - ").concat(sizing.scale0, ")") : sizing.scale400
  }, _defineProperty(_SIZE$default, $theme.direction === 'rtl' ? 'paddingRight' : 'paddingLeft', paddingLeft), _defineProperty(_SIZE$default, $theme.direction === 'rtl' ? 'paddingLeft' : 'paddingRight', 0), _SIZE$default)), _defineProperty(_SIZE$compact$SIZE$de2, _constants.SIZE.large, (_SIZE$large = {
    // `sizing.scale0` based on the multi value component (Tag) top and bottom margin
    paddingTop: $multi && !emptyValue ? "calc(".concat(sizing.scale550, " - ").concat(sizing.scale0, ")") : sizing.scale550,
    paddingBottom: $multi && !emptyValue ? "calc(".concat(sizing.scale550, " - ").concat(sizing.scale0, ")") : sizing.scale550
  }, _defineProperty(_SIZE$large, $theme.direction === 'rtl' ? 'paddingRight' : 'paddingLeft', $multi && !emptyValue ? "calc(".concat(paddingLeft, " - ").concat(sizing.scale0, ")") : paddingLeft), _defineProperty(_SIZE$large, $theme.direction === 'rtl' ? 'paddingLeft' : 'paddingRight', 0), _SIZE$large)), _SIZE$compact$SIZE$de2)[$size];
}

var StyledDropdownContainer = (0, _index.styled)('div', function (props) {
  return {
    width: "".concat(String(props.$width), "px")
  };
});
exports.StyledDropdownContainer = StyledDropdownContainer;
StyledDropdownContainer.displayName = "StyledDropdownContainer";
var StyledDropdown = _index2.StyledList;
exports.StyledDropdown = StyledDropdown;
var StyledDropdownListItem = _index2.StyledListItem;
exports.StyledDropdownListItem = StyledDropdownListItem;
var StyledOptionContent = (0, _index.styled)('div', function (props) {
  var $isHighlighted = props.$isHighlighted,
      $selected = props.$selected,
      $disabled = props.$disabled,
      $theme = props.$theme;
  return {
    cursor: $disabled ? 'not-allowed' : 'pointer',
    color: $selected && !$isHighlighted ? $theme.colors.menuFontSelected : null,
    fontWeight: $selected ? 'bold' : 'normal'
  };
});
exports.StyledOptionContent = StyledOptionContent;
StyledOptionContent.displayName = "StyledOptionContent";
var StyledRoot = (0, _index.styled)('div', function (props) {
  var typography = props.$theme.typography,
      $size = props.$size;
  return _objectSpread({}, getFont($size, typography), {
    boxSizing: 'border-box',
    position: 'relative'
  });
});
exports.StyledRoot = StyledRoot;
StyledRoot.displayName = "StyledRoot";

function getControlContainerColors($disabled, $isFocused, $isPseudoFocused, $positive, $error, colors) {
  if ($disabled) {
    return {
      color: colors.inputTextDisabled,
      borderColor: colors.inputFillDisabled,
      backgroundColor: colors.inputFillDisabled
    };
  }

  if ($isFocused || $isPseudoFocused) {
    return {
      color: colors.foreground,
      borderColor: colors.primary400,
      backgroundColor: colors.inputFillActive
    };
  }

  if ($error) {
    return {
      color: colors.foreground,
      borderColor: colors.inputBorderError,
      backgroundColor: colors.inputFillError
    };
  }

  if ($positive) {
    return {
      color: colors.foreground,
      borderColor: colors.inputBorderPositive,
      backgroundColor: colors.inputFillPositive
    };
  }

  return {
    color: colors.foreground,
    borderColor: colors.inputFill,
    backgroundColor: colors.inputFill
  };
}

var StyledControlContainer = (0, _index.styled)('div', function (props) {
  var $disabled = props.$disabled,
      $error = props.$error,
      $positive = props.$positive,
      $isFocused = props.$isFocused,
      $isPseudoFocused = props.$isPseudoFocused,
      $type = props.$type,
      $searchable = props.$searchable,
      _props$$theme = props.$theme,
      borders = _props$$theme.borders,
      colors = _props$$theme.colors,
      animation = _props$$theme.animation;
  return _objectSpread({
    borderTopLeftRadius: borders.inputBorderRadius,
    borderTopRightRadius: borders.inputBorderRadius,
    borderBottomRightRadius: borders.inputBorderRadius,
    borderBottomLeftRadius: borders.inputBorderRadius,
    boxSizing: 'border-box',
    overflow: 'hidden',
    width: '100%',
    display: 'flex',
    justifyContent: 'space-between',
    cursor: $disabled ? 'not-allowed' : $searchable || $type === _constants.TYPE.search ? 'text' : 'pointer',
    borderWidth: '2px',
    borderStyle: 'solid',
    transitionProperty: 'border, box-shadow, background-color',
    transitionDuration: animation.timing100,
    transitionTimingFunction: animation.easeOutCurve
  }, getControlContainerColors($disabled, $isFocused, $isPseudoFocused, $positive, $error, colors));
});
exports.StyledControlContainer = StyledControlContainer;
StyledControlContainer.displayName = "StyledControlContainer";
var StyledValueContainer = (0, _index.styled)('span', function (props) {
  var padding = getControlPadding(props);
  return _objectSpread({
    boxSizing: 'border-box',
    position: 'relative',
    flexGrow: 1,
    flexShrink: 1,
    flexBasis: '0%',
    display: 'flex',
    alignItems: 'center',
    flexWrap: 'wrap',
    overflow: 'hidden'
  }, padding);
});
exports.StyledValueContainer = StyledValueContainer;
StyledValueContainer.displayName = "StyledValueContainer";
var StyledPlaceholder = (0, _index.styled)('div', function (props) {
  var $disabled = props.$disabled,
      colors = props.$theme.colors;
  return _objectSpread({
    position: 'absolute',
    top: 0,
    bottom: 0,
    right: 0,
    left: 0,
    color: $disabled ? colors.inputTextDisabled : colors.foregroundAlt,
    maxWidth: '100%',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap'
  }, getControlPadding(props, true));
});
exports.StyledPlaceholder = StyledPlaceholder;
StyledPlaceholder.displayName = "StyledPlaceholder";
var StyledSingleValue = (0, _index.styled)('div', function (props) {
  var _objectSpread2;

  var $searchable = props.$searchable,
      $size = props.$size,
      $theme = props.$theme,
      typography = props.$theme.typography;
  var font = getFont($size, typography);
  return _objectSpread((_objectSpread2 = {
    lineHeight: !$searchable ? font.lineHeight : 'inherit',
    boxSizing: 'border-box',
    position: 'absolute',
    top: 0
  }, _defineProperty(_objectSpread2, $theme.direction === 'rtl' ? 'right' : 'left', 0), _defineProperty(_objectSpread2, "height", '100%'), _defineProperty(_objectSpread2, "maxWidth", '100%'), _objectSpread2), _util.ellipsisText, getControlPadding(props));
});
exports.StyledSingleValue = StyledSingleValue;
StyledSingleValue.displayName = "StyledSingleValue";
var StyledInputContainer = (0, _index.styled)('div', function (props) {
  var $size = props.$size,
      $searchable = props.$searchable,
      typography = props.$theme.typography;
  var font = getFont($size, typography);
  return {
    position: 'relative',
    display: 'inline-block',
    maxWidth: '100%',
    backgroundColor: 'transparent',
    border: 'none',
    boxShadow: 'none',
    boxSizing: 'border-box',
    outline: 'none',
    marginTop: 0,
    marginBottom: 0,
    marginLeft: 0,
    marginRight: 0,
    paddingTop: 0,
    paddingBottom: 0,
    paddingLeft: 0,
    paddingRight: 0,
    height: String(!$searchable ? font.lineHeight : 'auto'),
    maxHeight: font.lineHeight
  };
});
exports.StyledInputContainer = StyledInputContainer;
StyledInputContainer.displayName = "StyledInputContainer";
var StyledInput = (0, _index.styled)('input', function (props) {
  var typography = props.$theme.typography,
      $size = props.$size,
      $disabled = props.$disabled,
      $searchable = props.$searchable,
      $width = props.$width;
  return _objectSpread({}, getFont($size, typography), {
    boxSizing: 'content-box',
    width: $disabled || !$searchable ? '1px' : $width || '100%',
    maxWidth: '100%',
    background: 'transparent',
    border: 'none',
    boxShadow: 'none',
    display: 'inline-block',
    outline: 'none',
    marginTop: '0',
    marginBottom: '0',
    marginLeft: '0',
    marginRight: '0',
    paddingTop: '0',
    paddingBottom: '0',
    paddingLeft: '0',
    paddingRight: '0'
  });
});
exports.StyledInput = StyledInput;
StyledInput.displayName = "StyledInput";
var StyledInputSizer = (0, _index.styled)('div', function (_ref) {
  var _objectSpread3;

  var $size = _ref.$size,
      $theme = _ref.$theme,
      typography = _ref.$theme.typography;
  return _objectSpread({}, getFont($size, typography), (_objectSpread3 = {
    position: 'absolute',
    top: 0
  }, _defineProperty(_objectSpread3, $theme.direction === 'rtl' ? 'right' : 'left', 0), _defineProperty(_objectSpread3, "visibility", 'hidden'), _defineProperty(_objectSpread3, "height", 0), _defineProperty(_objectSpread3, "overflow", 'scroll'), _defineProperty(_objectSpread3, "whiteSpace", 'pre'), _objectSpread3));
});
exports.StyledInputSizer = StyledInputSizer;
StyledInputSizer.displayName = "StyledInputSizer";
var StyledIconsContainer = (0, _index.styled)('div', function (_ref2) {
  var $theme = _ref2.$theme,
      sizing = _ref2.$theme.sizing;
  return _defineProperty({
    boxSizing: 'border-box',
    position: 'relative',
    display: 'flex',
    flexShrink: 0,
    alignItems: 'center',
    alignSelf: 'stretch'
  }, $theme.direction === 'rtl' ? 'paddingLeft' : 'paddingRight', sizing.scale500);
});
exports.StyledIconsContainer = StyledIconsContainer;
StyledIconsContainer.displayName = "StyledIconsContainer";

function getSvgStyles(_ref4) {
  var $theme = _ref4.$theme;
  return {
    display: 'inline-block',
    fill: 'currentColor',
    color: 'currentColor',
    height: $theme.sizing.scale600,
    width: $theme.sizing.scale600
  };
}

var StyledSelectArrow = (0, _index.styled)('svg', function (props) {
  var $theme = props.$theme,
      $disabled = props.$disabled;
  var colors = $theme.colors;
  return _objectSpread({}, getSvgStyles({
    $theme: $theme
  }), {
    color: $disabled ? colors.inputTextDisabled : colors.foregroundAlt,
    cursor: $disabled ? 'not-allowed' : 'pointer'
  });
});
exports.StyledSelectArrow = StyledSelectArrow;
StyledSelectArrow.displayName = "StyledSelectArrow";
var StyledClearIcon = (0, _index.styled)('svg', function (props) {
  var $theme = props.$theme;
  var colors = $theme.colors;
  return _objectSpread({}, getSvgStyles({
    $theme: $theme
  }), {
    color: colors.foregroundAlt,
    cursor: 'pointer'
  });
});
exports.StyledClearIcon = StyledClearIcon;
StyledClearIcon.displayName = "StyledClearIcon";

var getLoadingIconStyles = function getLoadingIconStyles(props) {
  var $theme = props.$theme;
  var colors = $theme.colors;
  return _objectSpread({}, getSvgStyles({
    $theme: $theme
  }), {
    color: colors.foregroundAlt
  });
};

exports.getLoadingIconStyles = getLoadingIconStyles;
var StyledSearchIcon = (0, _index.styled)('div', function (props) {
  var _objectSpread4;

  var $disabled = props.$disabled,
      $theme = props.$theme;
  var colors = $theme.colors,
      sizing = $theme.sizing;
  return _objectSpread({}, getSvgStyles(props), (_objectSpread4 = {
    color: $disabled ? colors.inputTextDisabled : colors.foregroundAlt,
    cursor: $disabled ? 'not-allowed' : 'pointer',
    position: 'absolute'
  }, _defineProperty(_objectSpread4, $theme.direction === 'rtl' ? 'right' : 'left', sizing.scale500), _defineProperty(_objectSpread4, "display", 'flex'), _defineProperty(_objectSpread4, "alignItems", 'center'), _defineProperty(_objectSpread4, "height", '100%'), _objectSpread4));
});
exports.StyledSearchIcon = StyledSearchIcon;
StyledSearchIcon.displayName = "StyledSearchIcon";
