// baseui/esm/themes/light-theme-with-move.js
function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

/*
Copyright (c) 2018-2019 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
import createTheme from './creator.js';
import { primitives as lightThemePrimitives } from './light-theme-primitives.js';
var primaryFontFamily = 'UberMoveText, system-ui, "Helvetica Neue", Helvetica, Arial, sans-serif';
var secondaryFontFamily = 'UberMove, UberMoveText, system-ui, "Helvetica Neue", Helvetica, Arial, sans-serif';
export var LightThemeMove = createTheme(_objectSpread({}, lightThemePrimitives, {
  primaryFontFamily: primaryFontFamily
}), {
  name: 'light-theme-with-move',
  typography: {
    font1100: {
      fontFamily: secondaryFontFamily
    }
  }
});
